### ISSUE TITLE
TensorboardLogger - tag/prefix for weights, grads and optimizer logging

### ISSUE THREAD

**Issue by Bibonaut** [pytorch/ignite#598]
### Problem
I have to log multiple models/optimizers with TensorboardLogger and I want them to be displayed in tensorboard in separate *subpanels* with distinct names/tags. 
At the moment, it is not possible to provide a custom tag or prefix for some handlers that log model parameters, because the tag is deduced from `model.named_parameters()`. 

As a consequence, parameters of multiple models are plottet to one graph.

### Solution
The affected handlers allow passing a custom prefix/tag to the function that is prepended to the deduced tag. 

E.g. for `GradsHistHandler` (snippet):
```python
logger.writer.add_histogram(tag="grads/{}".format(name),
                                        values=p.grad.detach().cpu().numpy(),
                                        global_step=global_step)
```

It can be changed to 
```python
logger.writer.add_histogram(tag=self.prefix + "grads/{}".format(name),
                                        values=p.grad.detach().cpu().numpy(),
                                        global_step=global_step)
```

Affected handlers in `ignite.contrib.handlers.tensorboard_logger`:
- ~~`OptimizerParamsHandler`~~
- `WeightsScalarHandler`
- `WeightsHistHandler`
- `GradsScalarHandler`
- `GradsHistHandler`

What do you think? 
If wanted, I can work on this.

**Comment by vfdev-5** [comment]
@Bibonaut I think we had something similar with `OptimizerParamsHandler` and we added `tag` in https://github.com/pytorch/ignite/pull/502

I agree that we may need to add `tag` to other handlers too.

So, yes, if you can send a PR on this, it would be very helpful !

**Comment by Bibonaut** [comment]
@vfdev-5 Thank you for the hint with `OptimizerParamsHandler`.  I see that in #502 the `tag` property was also introduced in `BaseOptimizerParamsHandler`.  So I would do it the same way for gradients and weights handlers and add `tag` property to their base classes. Do you agree?

**Comment by vfdev-5** [comment]
@Bibonaut sounds good, please go ahead 👍
